froide/helper/breadcrumbs.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+import logging
+from dataclasses import dataclass
+from typing import Generator, Optional, Sequence, Tuple, Union
+
+from django.template.context import Context
+from django.urls import NoReverseMatch, reverse
+from django.views import View
+
+logger = logging.getLogger(__name__)
+
+BreadcrumbTuple = Tuple[str, Union[str, None]]
+"""
+First item: link text
+Second item: link, or None
+"""
+
+BreadcrumbItems = Sequence[Union[str, BreadcrumbTuple]]
+
+
+class BreadcrumbView(View):
+    """
+    Your view must either provide a `breadcrumbs` attribute, or a
+    `get_breadcrumbs` method.
+
+    The `breadcrumbs` attribute is a sequence of strings or tuples,
+    e.g. ["Just text", ("A link", "url-pattern-name")].
+    The first element of the tuple specifies the link text, the second one
+    will be REVERSED to the link, so you can provide path names.
+
+    Using the `get_breadcrumbs` method, you can provide breadcrumbs as
+    above, with the difference that urls WILL NOT BE REVERSED.
+    You can also directly return a `Breadcrumbs` instance.
+
+    Example implementation:
+
+    class MyView(BreadcrumbView):
+        def get_breadcrumbs():
+            return [("Foo", "/foo/"), "Bar"]
+    """
+
+    breadcrumbs: Optional[Sequence]
+
+    def get_breadcrumbs(self) -> Union[BreadcrumbItems, Breadcrumbs]:
+        raise NotImplementedError("No breadcrumb provider implemented")
```

Comment:
To simplify the interface the the view, we could let `get_breadcrumbs` return `breadcrumbs` if set and raise otherwise, similar to what django does with get_queryset on generic views [here](https://github.com/django/django/blob/86e13843c2ab510fba1de84588efe7fd03555531/django/views/generic/detail.py#L61)